### DISCUSSION TITLE
Use seven-character short SHA in PR generator

### DISCUSSION THREAD

**Discussion by mseiwald** [argoproj/argo-cd#10194] [Q&A] [answered]
Hi all,

Recently the variable `head_short_sha` [was added](https://github.com/argoproj/argo-cd/pull/9669) to the PR generator. This functionality uses the first 8 characters of the SHA which is unfortunate since many tools including `git` itself will default to 7 characters.

I was wondering why this was done in that way and if there are any workarounds (Go templates seem to be not supported) to get a 7-character short SHA in PR generator application sets.

Thanks!

**Comment by crenshaw-dev** [comment, marked as answer]
As far as why, looks like it was just based on GitLab's standard: https://github.com/argoproj/argo-cd/pull/9669#issuecomment-1159166285

I think in 2.5 you'll be able to use Go Text Templating to truncate the SHA to the length you want. https://github.com/argoproj/argo-cd/pull/10026

For the moment, I don't think there's a workaround from within ApplicationSet.
